playwright/_impl/_browser_context.py

```diff
@@ -248,15 +248,18 @@ def set_default_navigation_timeout(self, timeout: float) -> None:
     def _set_default_navigation_timeout_impl(self, timeout: Optional[float]) -> None:
         self._timeout_settings.set_default_navigation_timeout(timeout)
         self._channel.send_no_reply(
-            "setDefaultNavigationTimeoutNoReply", dict(timeout=timeout)
+            "setDefaultNavigationTimeoutNoReply",
+            {} if timeout is None else {"timeout": timeout},
```

Comment:
The issue was that on protocol level we don't allow timeout to be null.

Reply:
Should we consider serializing None as undefined or too many things break?

Reply:
too many things would break I believe!